libnavigation-base/src/main/java/com/mapbox/navigation/base/internal/route/NavigationRouteEx.kt

```diff
@@ -226,14 +231,23 @@ private fun buildNewWaypoints(
     }
 }
 
-private fun DirectionsRoute.Builder.updateRouteDurationBasedOnLegsDuration(
-    updateLegs: List<RouteLeg>?
+private fun DirectionsRoute.Builder.updateRouteDurationBasedOnLegsDurationAndChargeTime(
+    updateLegs: List<RouteLeg>?,
+    waypoints: List<DirectionsWaypoint?>?,
 ): DirectionsRoute.Builder {
     updateLegs ?: return this
     var result = 0.0
     for (leg in updateLegs) {
         result += leg.duration() ?: return this
     }
+    waypoints?.forEach { waypoint ->
+        waypoint?.unrecognizedJsonProperties
+            ?.get("metadata")
+            ?.asJsonObject
+            ?.get("charge_time")?.asDouble?.let { chargeTime ->
+                result += chargeTime
+            }
```

Comment:
Do you think it would be useful to introduce `Waypoint.chargeTime function as an internal API?

Reply:
We can look into it if we actually need it in more than one place in the production code.